src/test/java/com/example/animalshelter/controller/AnimalControllerIT.java

```diff
@@ -68,4 +81,47 @@ public void shouldReturnPositiveHttpResponseWithEmptyAnimalObjectWhenAnimalIsEmp
                 .andDo(print())
                 .andExpect(status().isBadRequest());
     }
+
+    @Test
+    public void shouldReturnPositiveHttpResponseWhenDeletedAnimalSuccessfully() throws Exception {
+        String asJson = new ObjectMapper().writeValueAsString(DUMMY_ANIMAL);
+
+        this.mockMvc.perform(
+                post("/animals")
+                        .content(asJson)
+                        .contentType(MediaType.APPLICATION_JSON));
+
+        this.mockMvc.perform(
+                        delete("/animals/{id}", DUMMY_ANIMAL.getId()))
+                .andDo(print())
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenIdIsNegative() throws Exception {
```

Comment:
Now when I see the test I would definately go for bad request because of validation error.

Reply:
Done, now test check if it is bad request in that case.